tensornetwork/network_components.py

```diff
@@ -1040,6 +1065,31 @@ def update_axis(self, old_axis: int, old_node: AbstractNode, new_axis: int,
                            self, old_node, old_axis, self.node1, self.axis1,
                            self.node2, self.axis2))
 
+  @property
+  def nodes(self) -> Tuple[AbstractNode]:
+    if self.is_disabled:
+      raise ValueError(
+          'Edge has been disabled, accessing nodes is no longer possible')
+    return tuple(self._nodes)
+
+  @nodes.setter
+  def nodes(self, nodes: Iterable[AbstractNode]) -> None:
+    if self.is_disabled:
+      raise ValueError(
+          'Edge has been disabled, setting nodes is no longer possible')
+    if not all(map(lambda x: isinstance(x, AbstractNode) or x is None), nodes):
+      raise TypeError("All node values should be Node type")
+    self._nodes = list(nodes)
+
+  def set_node(self, node_pos, node) -> None:
+    if node_pos < 0 or node_pos >= len(self._nodes):
+      raise ValueError(
+          f"node_pos {node_pos} is outside the range 0 - {len(self._nodes) - 1}"
+          )
+    if not (isinstance(node, AbstractNode) or node is None):
+      raise TypeError("node must be of type AbstractNode")
+    self._nodes[node_pos] = node
```

Comment:
I'm not really sure how I feel about needing to know the position of the node to change it. The position in the list is really arbitrary.

What do you think about having a `{node: [axes]}` dictionary instead? That way we won't need to worry about the positions in the list.

Reply:
Yeah I think that makes more sense. I definitely agree that having to know the position is too clunky.

But it does seem like there is still value in having some concept of node order. Like it seems edge.nodes[0] should always be the same. And since almost all of the edge operations are build around the concept of node0, node1 it seems like being able to to maintain the concept of the first node and the second node could be helpful.

Perhaps use a dict and just rely on the ordering of python 3.6+ dicts when returning edge.nodes and edge.axes is enough?

Reply:
I guess I'm wondering are there any operations that would really ever directly set a single axis or node besides the `update_axis` method? Cause if not then we don't need to worry about order at all.